### ISSUE TITLE
Exclude m/z values during GCMS spectral deconvolution

### ISSUE THREAD

**Issue by NthlLcrmp** [mzmine/mzmine#1145]
Dear all,

I would like to know how the m/z exclusion list works during spectral deconvolution of GC/MS data with Hierarchical clustering on MZmine 3.3.0.

When selecting the model peak, I would like to be able to exclude m/z 73 and 147 as possible choices and select the most intense m/z after these two. I tried entering integer m/z values (e.g. 73, 147) or ranges (e.g. 72-75, 146-148) but these m/z are still selected as model peak as soon as the choice is based on intensity.

Thank you for your help

Best,
Nathalie

**Comment by robinschmid** [comment]
Maybe @asmirn1 or @tnguy271 could comment they are the experts on GC-MS data.

**Comment by asmirn1** [comment]
@NthlLcrmp Could you share your project file so I could test it?

The m/z exclusion list should work with any choice of Model Peak. What parameter do you use for the spectral deconvolution?

**Comment by NthlLcrmp** [comment]
@asmirn1 Thank you for taking a look at the problem.

Here is a link with a test project file:
https://filesender.renater.fr/?s=download&token=4117541f-57d1-4e31-980b-4b06f999e11b

The parameters used for spectral deconvolution were 

![image](https://user-images.githubusercontent.com/114736047/219854098-2342bdf3-7bb9-4e0f-9b02-0403133899e0.png)

I also try with m/z ranges 
![image](https://user-images.githubusercontent.com/114736047/219854222-e1247d85-4924-4160-8344-6c7466d7aa4e.png)

I am not really confident in my choices, I may have made mistakes.
Both cases produce the same results with m/z 73 as model peak : 

![image](https://user-images.githubusercontent.com/114736047/219855026-997afab0-058e-4dda-bf47-ef971caf8992.png)

**Comment by asmirn1** [comment]
@NthlLcrmp 

We've fixed this issue in the development release. Could you try to install it and let us know if it works for you?

https://github.com/mzmine/mzmine3/releases/tag/Development-release

**Comment by NthlLcrmp** [comment]
Thank you @asmirn1 for the update. 

The exclusion of m/z values works fine with this version. 
However, an m/z range must be provided to exclude the unwanted m/z. When choosing only the m/z value 73 the feature list still contains m/z 73 :
![image](https://user-images.githubusercontent.com/114736047/221799957-619d021b-9ca4-46f2-bfa2-3d89bd3e058a.png)

What is the tolerance on the m/z value to exclude?

**Comment by NthlLcrmp** [comment]
@asmirn1 I agree that a single parameter with ranges is simpler.  It might be useful to point out in the tooltip that a range is needed for non-integer m/z values.

I have a related question regarding the number of significant digits on the m/z value of the model peak.  
I process low resolution data with one significant digit on the m/z values. When extracting EICs I use a scan to scan accuracy of 0.5 m/z or 0 ppm and in the resulting feature list the m/z has a precision of 4 significant digits which is maintained throughout the data processing. 
Is it possible to produce m/z values with fewer significant digits?
Is this due to a poor choice of parameters?
